src/reducer.js

```diff
@@ -0,0 +1,247 @@
+import {
+    formatServerError,
+    formatGraphQLError,
+    dispatchMutationReq,
+    dispatchMutationResp,
+    dispatchMutationErr,
+    parseData,
+    pageInfo,
+    decodeId,
+} from "@openimis/fe-core";
+import {REQUEST, SUCCESS, ERROR, CLEAR} from "./util/action-type";
+
+export const ACTION_TYPE = {
+    MUTATION: "BENEFIT_PLAN_MUTATION",
+    SEARCH_BENEFIT_PLANS: "BENEFIT_PLAN_BENEFIT_PLANS",
+    GET_BENEFIT_PLAN: "BENEFIT_PLAN_BENEFIT_PLAN",
+    DELETE_BENEFIT_PLAN: "BENEFIT_PLAN_DELETE_BENEFIT_PLAN",
+    UPDATE_BENEFIT_PLAN: "BENEFIT_PLAN_UPDATE_BENEFIT_PLAN",
+    BENEFIT_PLAN_CODE_FIELDS_VALIDATION: "BENEFIT_PLAN_CODE_FIELDS_VALIDATION",
+    BENEFIT_PLAN_NAME_FIELDS_VALIDATION: "BENEFIT_PLAN_NAME_FIELDS_VALIDATION",
+    BENEFIT_PLAN_CODE_SET_VALID: "BENEFIT_PLAN_CODE_SET_VALID",
+    BENEFIT_PLAN_NAME_SET_VALID: "BENEFIT_PLAN_NAME_SET_VALID"
+};
+
+function reducer(
+    state = {
+        submittingMutation: false,
+        mutation: {},
+        fetchingBenefitPlans: false,
+        errorBenefitPlans: null,
+        fetchedBenefitPlans: false,
+        benefitPlans: [],
+        benefitPlansPageInfo: {},
+        benefitPlansTotalCount: 0,
+        fetchingBenefitPlan: false,
+        errorBenefitPlan: null,
+        fetchedBenefitPlan: false,
+        benefitPlan: null
+    },
+    action,
+) {
+    switch (action.type) {
+        case REQUEST(ACTION_TYPE.SEARCH_BENEFIT_PLANS):
+            return {
+                ...state,
+                fetchingBenefitPlans: true,
+                fetchedBenefitPlans: false,
+                benefitPlans: [],
+                benefitPlansPageInfo: {},
+                benefitPlansTotalCount: 0,
+                errorBenefitPlans: null,
+            };
+        case REQUEST(ACTION_TYPE.GET_BENEFIT_PLAN):
+            return {
+                ...state,
+                fetchingBenefitPlan: true,
+                fetchedBenefitPlan: false,
+                benefitPlan: null,
+            };
+        case SUCCESS(ACTION_TYPE.SEARCH_BENEFIT_PLANS):
+            return {
+                ...state,
+                fetchingBenefitPlans: false,
+                fetchedBenefitPlans: true,
+                benefitPlans: parseData(action.payload.data.benefitPlan)?.map((benefitPlan) => {
+                    let res = ({
```

Comment:
Can you rename this variable to 'response'? Could be better to have full name of such variable.

Reply:
DONE